File: JavaScript/8js 1/countSheep.js
/*
|--------------------------------------------------------------------------
| If you can't sleep, just count sheep!!
|--------------------------------------------------------------------------
|
*/

function countSheep(num) {
   let res = '';
   for (let i = 1; i <= num; i++) {
      res += `${i} sheep...`;
   }
   return res
}


const countSheep = n => [...Array(n).keys()].map(x => `${x+1} sheep...`).join ``


const countSheep = length =>
   Array.from({
      length
   }, (_, i) => ++i + ' sheep...').join('')


const countSheep = function (n) {
   return [...Array(n)].map((_, i) => i + 1 + ' sheep...').join ``
}


console.log(countSheep(1), "1 sheep...");
console.log(countSheep(2), "1 sheep...2 sheep...");
console.log(countSheep(3), "1 sheep...2 sheep...3 sheep...");

File: Python/8py 1/count_sheep.py
"""
|--------------------------------------------------------------------------
| If you can't sleep, just count sheep!!
|--------------------------------------------------------------------------
|
"""

def count_sheep(n):
   res = ''
   for i in range(1, n+1):
      res += f'{i} sheep...'
   return res


def count_sheep(n):
   return ''.join(f"{i} sheep..." for i in range(1,n+1))


def count_sheep(n):
   return "".join("%d sheep..." % (i + 1) for i in range(n))


def count_sheep(n):
   return "".join("{} sheep...".format(i) for i in range(1, n+1))


def count_sheep(n):
   return ''.join(f'{x+1} sheep...' for x in range(n))


print(count_sheep(1), "1 sheep...");
print(count_sheep(2), "1 sheep...2 sheep...")
print(count_sheep(3), "1 sheep...2 sheep...3 sheep...")